src/librustc/mir/mod.rs

```diff
@@ -471,6 +464,7 @@ pub enum TerminatorKind<'tcx> {
 
         /// Possible values. The locations to branch to in each case
         /// are found in the corresponding indices from the `targets` vector.
+        // FIXME: ConstVal doesn’t quite make any sense here? Its a Switch*Int*.
         values: Vec<ConstVal>,
 
         /// Possible branch sites. The length of this vector should be
```

Comment:
It seems like we may want to adjust `SwitchInt` here -- perhaps you did so but I didn't notice, but it so this comment is out of date. The current comment suggests you always need a "fall-through" case, but we don't want that now, right? Although I guess that we could code it up by loading the set of discriminants (let's call them `D[i]` for `i = 1...n`) and then generating branches for `D[i]` where `i in 1...n-1`, and using the fall through case for `D[n]`. That is probably better, actually, as I'd prefer to keep this uniform.